packages/turf-boolean-within/index.js

```diff
@@ -0,0 +1,218 @@
+var inside = require('@turf/inside');
+var calcBbox = require('@turf/bbox');
+var invariant = require('@turf/invariant');
+var isPointOnLine = require('@turf/boolean-point-on-line');
+var getGeom = invariant.getGeom;
+var getGeomType = invariant.getGeomType;
+
+/**
+ * Boolean-within returns true if the first geometry is completely within the second geometry.
+ * The interiors of both geometries must intersect and, the interior and boundary of the primary (geometry a)
+ * must not intersect the exterior of the secondary (geometry b).
+ * Boolean-within returns the exact opposite result of the `@turf/boolean-contains`.
+ *
+ * @name booleanWith
+ * @param {Geometry|Feature<any>} feature1 GeoJSON Feature or Geometry
+ * @param {Geometry|Feature<any>} feature2 GeoJSON Feature or Geometry
+ * @returns {boolean} true/false
+ * @example
+ * const line = turf.lineString([[1, 1], [1, 2], [1, 3], [1, 4]]);
+ * const point = turf.point([1, 2]);
+ *
+ * turf.booleanWithin(point, line);
+ * //=true
+ */
+module.exports = function (feature1, feature2) {
+    var type1 = getGeomType(feature1);
+    var type2 = getGeomType(feature2);
+    var geom1 = getGeom(feature1);
+    var geom2 = getGeom(feature2);
+
+    switch (type1) {
+    case 'Point':
+        switch (type2) {
+        case 'MultiPoint':
+            return isPointInMultiPoint(geom1, geom2);
+        case 'LineString':
+            return isPointOnLine(geom1, geom2, true);
+        case 'Polygon':
+            return inside(geom1, geom2, true);
+        default:
+            throw new Error('feature2 ' + type2 + ' geometry not supported');
```

Comment:
Instead of throwing an error, it would be better to throw `false` if Point cannot be within another Point.

Reply:
Hmmm this is the pattern we've followed for all the other boolean modules. I'd suggest if we want to change our approach for one then we should do it for all of them. Perhaps we should do this as a `v5` change?

Reply:
Actually, no might be fine... since some of the `within` combinations don't actually make sense, this is good for now. 👍